### ISSUE TITLE
Quanteda 1.4.1 loses tokens when input data is big

### ISSUE THREAD

**Issue by Astelix** [quanteda/quanteda#1640]
I got different results from several analyses with Quanteda 1.3.4 and 1.4.1. The problem seems to stem from version 1.4.1 seems to loose tokens when the input data is big.

To reproduce the following example you need my data file (which I dont't want to upload here).

``` r
sessionInfo()
#> R version 3.5.1 (2018-07-02)
#> Platform: x86_64-w64-mingw32/x64 (64-bit)
#> Running under: Windows 8.1 x64 (build 9600)
#> 
#> Matrix products: default
#> 
#> locale:
#> [1] LC_COLLATE=German_Germany.1252  LC_CTYPE=German_Germany.1252   
#> [3] LC_MONETARY=German_Germany.1252 LC_NUMERIC=C                   
#> [5] LC_TIME=German_Germany.1252    
#> 
#> attached base packages:
#> [1] stats     graphics  grDevices utils     datasets  methods   base     
#> 
#> loaded via a namespace (and not attached):
#>  [1] compiler_3.5.1  magrittr_1.5    tools_3.5.1     htmltools_0.3.6
#>  [5] yaml_2.2.0      Rcpp_1.0.0      stringi_1.3.1   rmarkdown_1.11 
#>  [9] highr_0.7       knitr_1.21      stringr_1.4.0   xfun_0.5       
#> [13] digest_0.6.18   evaluate_0.13

library(tidyverse)
#> Warning: Paket 'tidyverse' wurde unter R Version 3.5.2 erstellt
#> Warning: Paket 'ggplot2' wurde unter R Version 3.5.2 erstellt
#> Warning: Paket 'tibble' wurde unter R Version 3.5.2 erstellt
#> Warning: Paket 'tidyr' wurde unter R Version 3.5.2 erstellt
#> Warning: Paket 'readr' wurde unter R Version 3.5.2 erstellt
#> Warning: Paket 'dplyr' wurde unter R Version 3.5.2 erstellt
#> Warning: Paket 'stringr' wurde unter R Version 3.5.2 erstellt
#> Warning: Paket 'forcats' wurde unter R Version 3.5.2 erstellt
library(quanteda)
#> Warning: Paket 'quanteda' wurde unter R Version 3.5.2 erstellt
#> Package version: 1.4.1
#> Parallel computing: 2 of 8 threads used.
#> See https://quanteda.io for tutorials and examples.
#> 
#> Attache Paket: 'quanteda'
#> The following object is masked from 'package:utils':
#> 
#>     View

setwd("xxxxxxxxxxxxxxxxxx")
load(file="all_articles.rdata")
#loads all_articles with 11366 obs and 6 variables and 6.5 million tokens

test_cases <- all_articles %>% filter(str_detect(text,"Khashoggi|Gelbweste"))
#reduced version with only 204 cases 6 variables ans 125000 tokens

corp_full <- corpus(all_articles, text_field = "text")
full_dfm  <- dfm(corp_full)

sum(full_dfm[,"khashoggi"])
#> [1] 29
sum(full_dfm[,"gelbwesten"])
#> [1] 125


#The reduced corpus gets more hits
#This is clearly wrong
#The result in the reduced version is correct. I Checked that with KWIC.
#In Quanteda version 1.3.4 the results for the full and the reduced version are identical
#Quanteda 1.4.1 is loosing tokens

corp_test <- corpus(test_cases, text_field = "text")
test_dfm  <- dfm(corp_test)

sum(test_dfm[,"khashoggi"])
#> [1] 751
sum(test_dfm[,"gelbwesten"])
#> [1] 507
```

<sup>Created on 2019-03-07 by the [reprex package](https://reprex.tidyverse.org) (v0.2.1)</sup>

**Comment by koheiw** [comment]
@Astelix  we cannot do anything unless you send us the data or replicate using the data in the package.

**Comment by Astelix** [comment]
OK here is the dataset
[TSArticles.zip](https://github.com/quanteda/quanteda/files/2940319/TSArticles.zip)
Please dont't leave it on the website

**Comment by Astelix** [comment]
I think the probnlem is the same as in "dfm_group fails on Windows when groups are non-ASCII labels" (#1639) which I mailed to KBENOIT yesterday. The problem seams to disappear after unicode normalizing the text with:

all_articles$text <-  stringi::stri_trans_nfc(all_articles$text)

I never knew that something like this existed. ;)

**Comment by koheiw** [comment]
Can you try this? I do not see any problem on my Windows and Linux.
```r
require(quanteda)
sessionInfo()
load("C:\\Users\\Kohei\\Downloads\\TSArticles\\TSArticles.rdata")
corp <- corpus(all_articles, text_field = "text")
dfmt <- dfm(corp)
dfmt_sub <- dfm(head(corp, 5000))
feat <- names(topfeatures(dfmt_sub, 30))
all(colSums(dfmt[,feat]) >= colSums(dfmt_sub[,feat])) # should be TRUE
```

**Comment by Astelix** [comment]
It returns true.

If I understand your code right, you are checking, if the count of the topfeatures is always smaller in the first 5000 docs of the corpus than in the complete corpus. But this doesn't show everything. The problems are not with topfeatures and not at the beginning of the file.

Try this:


colSums(dfmt[,"khashoggi"])  #returns 29 here
dfmt_sub <- dfm(tail(corp, 5000))
colSums(dfmt_sub[,"khashoggi"]) #returns 751 here


### This will correct it

all_articles$text <-  stringi::stri_trans_nfc(all_articles$text)
corp <- corpus(all_articles, text_field = "text")
dfmt <- dfm(corp)
colSums(dfmt[,"khashoggi"])  #returns 751 here

**Comment by koheiw** [comment]
Then adding the Unicode normalization function should fix the issue `corpus.character()`.

**Comment by Astelix** [comment]
The problem should be fixed for text and docvars (see #1639) that it works automatically like my old version 1.3.9 again
